Tweak left-panel times

cadeban opened the pull request:
## Description
Changes:
- Move gauges into Electricity Mix section
  - because they are hourly-specific
- Remove leftPanel-scoped time
  - because the leftPanel contains different timeframes
- Add subtitles to timeseries charts with time range

[Designs](https://www.figma.com/design/O1DtS6UyZHSH4IymG8Ufsw/%F0%9F%8E%B9-Design-Hack?node-id=6855-31802&t=aj4ESiqtUZ3AWzSi-0)

TODO:
- [x] Add tests
- [x] Use `formatRange`

### Preview
<img width="270" alt="Screenshot 2025-03-27 at 10 04 17" src="https://github.com/user-attachments/assets/f62c79ec-1659-4c30-93d6-86f2fb001aee" />

<img width="270" alt="Screenshot 2025-03-27 at 10 04 37" src="https://github.com/user-attachments/assets/b6c033ab-6515-44d9-a4d9-43ce418eb946" />

### Double check
- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

cadeban commented:
Hey @Alportan, mind taking a look at the branch when you have time? 🙏

Alportan commented:
> Hey @Alportan, mind taking a look at the branch when you have time? 🙏
> 
> Will hold off on requesting a review until we've completed user testing :)

Looking good overall! 😎 

Tiny nitpick, but could we have some more padding under the title section of the cards? ↕️ In the editor it seems like it uses 0.125rem now, could we make it 0.5rem instead? 🙏 
<img width="411" alt="image" src="https://github.com/user-attachments/assets/924adc54-c167-410c-bd4d-197f7f519910" />

Also, it does look a bit empty without a date/time subtitle for the other charts similar to what @VIKTORVAV99 expressed in Figma 🤔 I wonder if it would feel super redundant to have those dates in 2 places (subheader and x-axis labels) but I guess user testing might help with that.

VIKTORVAV99 commented:
<!-- linear:replyTo:2757246504 -->

We can leave those dates for now and explore it a bit in the future, but 0.5 rem (8 px) sounds like a lot, would not 0.25 be better? (4px)

cadeban commented:
Added in the date ranges for the timeseries charts and added pb-2 to the chart headers:
<img width="284" alt="Screenshot 2025-03-27 at 16 42 26" src="https://github.com/user-attachments/assets/8f5e1b46-f08a-41b3-bfb7-c075d93a5176" />

<img width="278" alt="Screenshot 2025-03-27 at 16 42 07" src="https://github.com/user-attachments/assets/0703af52-403a-433f-a9fc-810c84a73de0" />

<img width="285" alt="Screenshot 2025-03-27 at 16 41 45" src="https://github.com/user-attachments/assets/bab02a76-b947-4fec-9e73-dcac67b065f8" />

Alportan commented:
> Added in the date ranges for the timeseries charts and added pb-2 to the chart headers: <img alt="Screenshot 2025-03-27 at 16 42 26" width="284" src="https://private-user-images.githubusercontent.com/7953060/427644776-8f5e1b46-f08a-41b3-bfb7-c075d93a5176.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxNTIxMDYsIm5iZiI6MTc0MzE1MTgwNiwicGF0aCI6Ii83OTUzMDYwLzQyNzY0NDc3Ni04ZjVlMWI0Ni1mMDhhLTQxYjMtYmZiNy1jMDc1ZDkzYTUxNzYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjhUMDg1MDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWY2ODE1OTk0Yzg3OWQ5ZDI1NTYzYzQ3ODg2ZDhlMjYxZDY1ZGFiNWYxMTczMThlMTM5Nzc3YmVjNmUwZGUyNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.JMT45yYaMjJYC-9ZUm_ZZra0-GtY_yzS0q_xscQgej0">
> 
> <img alt="Screenshot 2025-03-27 at 16 42 07" width="278" src="https://private-user-images.githubusercontent.com/7953060/427644770-0703af52-403a-433f-a9fc-810c84a73de0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxNTIxMDYsIm5iZiI6MTc0MzE1MTgwNiwicGF0aCI6Ii83OTUzMDYwLzQyNzY0NDc3MC0wNzAzYWY1Mi00MDNhLTQzM2YtYTlmYy04MTBjODRhNzNkZTAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjhUMDg1MDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2ZkNmExMGVjODQwYWViODRhZDcyM2FjMWI1OTIzZTRmNDVlYTk1NWRmZGQxZjYwNTk3ZTRmM2QzNTU1MWUzZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cX6zhxu2YSvhdZE8XYigOQTRbcwqBoqPnT78kYW9LTA"> <img alt="Screenshot 2025-03-27 at 16 41 45" width="285" src="https://private-user-images.githubusercontent.com/7953060/427644764-bab02a76-b947-4fec-9e73-dcac67b065f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxNTIxMDYsIm5iZiI6MTc0MzE1MTgwNiwicGF0aCI6Ii83OTUzMDYwLzQyNzY0NDc2NC1iYWIwMmE3Ni1iOTQ3LTRmZWMtOWU3My1kY2FjNjdiMDY1ZjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjhUMDg1MDA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTBlNWZhN2QxNmY0MGRmNWY4ZmEzODBkYTA4ZWMyMzAwZDE3NTQzODRkMzhhMTMwM2IyMGU2YzUzM2FlZmViYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.VzkMg95hF_MH5XMxGb-gdh4XahJePPw2UK5vEfNFAsc">

Looking good! 😎 The 0.8rem was a quick inspector hack I tried, but this works like a charm 🙌

VIKTORVAV99 commented on web/src/features/charts/ChartTitle.tsx:
```diff
@@ -45,7 +48,26 @@ export function ChartTitle({
+  const { i18n } = useTranslation();
+  return (
+    <p className="text-xs text-neutral-600 dark:text-neutral-300">
+      {getDateRange(i18n.language, datetimes, timeRange)}
```
```suggestion
      {getDateRange(i18n.languages[0], datetimes, timeRange)}
```
Language is the resolved language but languages[0] is the used language. They don't always match if the resolved language don't exists as it falls back to english.

cadeban replied:
Ahhh whoops, good catch!

VIKTORVAV99 reviewed (approved):
LGTM! 🎉

Just one tiny suggestion.
